src/Workspaces/Core/Portable/CodeFixes/FixAllOccurrences/FixAllContext.SpanBasedDiagnosticProvider.cs

```diff
@@ -0,0 +1,41 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Internal.Log;
+using Microsoft.CodeAnalysis.Text;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.CodeFixes
+{
+    /// <summary>
+    /// Context for "Fix all occurrences" code fixes provided by a <see cref="FixAllProvider"/>.
+    /// </summary>
+    public partial class FixAllContext
```

Comment:
Make Internal?

Reply:
@jmarolf This type is already public. Note that this PR is not for FixAll support for code refactorings that we discussed last week. This PR is extending the existing FixAll feature and public APIs for diagnostics/code fixes to add new fix all scopes.